### ISSUE TITLE
Support optional nonce attribute in /token and /authorize 

### ISSUE THREAD

**Issue by kishansagathiya** [fabric8-services/fabric8-auth#268]
the `api/authorize` and `api/token` endpoints should support the optional `nonce` attribute, as described in the specification here:
http://openid.net/specs/openid-connect-core-1_0.html#HybridAuthRequest

Required to be compatible with openshiftio-che

**Comment by kishansagathiya** [comment]
@davidfestal After going through the specs, I do not see `/token` supporting nonce. Are you sure `/token` should support nonce? Any reference?

**Comment by davidfestal** [comment]
see here: http://openid.net/specs/openid-connect-core-1_0.html#NonceNotes

and also here:
http://openid.net/specs/openid-connect-core-1_0.html#ImplicitAuthRequest
http://openid.net/specs/openid-connect-core-1_0.html#ImplicitIDToken
http://openid.net/specs/openid-connect-core-1_0.html#ImplicitIDTValidation

So as far as I understand either the returned token should contain a "nonce" claim with the same value as the one passed to the initial auth request.

**Comment by kishansagathiya** [comment]
@davidfestal Yes you are right. But it is for `response_type=id_token`, we are using `response_type=code`. References you sent are for implicit flow. Our APIs follow, authorization code flow. I don't think we would need to support nonce. Do you still think we need this?

**Comment by kishansagathiya** [comment]
@davidfestal My bad, Just saw an example of `response_type=code` with nonce in Appendix A.1. I will start working on this.

**Comment by kishansagathiya** [comment]
@davidfestal So no nonce for `/token`.
However nonce will be passed to  `/authorize` and on receiving token with `/token` nonce would be in id_token.
So now our OauthToken will be {access_token, expires_in, token_type, refresh_token, id_token}. And it is upto the client service to check that nonce received with id_token is same as the one sent with `/authorize`.
Though in didn't see this, but I assume nonce would also be unique just like state. We would probably need to store nonce somewhere in the database.
I am also tagging @alexeykazakov to confirm the flow.

**Comment by alexeykazakov** [comment]
I guess we can store nonce in the state table.

**Comment by sbryzak** [comment]
Still relevant as a low priority as of Jan 2019.
